src/components/pages/home/components/BrightConnectionModal/brightConnectionModal.tsx

```diff
@@ -1,21 +1,43 @@
 import * as React from 'react';
-import { useContext, useMemo } from 'react';
+import { useCallback, useContext, useMemo } from 'react';
 import { Text } from 'components/basic/Text/text.style';
 import {
   BrightConnectionModalWrapper,
   CopyLink,
 } from 'components/pages/home/components/BrightConnectionModal/brightConnectionModal.style';
 import { UserProfileContext } from '../../../../../hooks/useUserProfile';
+
+import { SecondaryButton } from 'components/basic/Button/button';
+
+import { BrightIdVerificationStatus } from 'types';
+
 import { getVerificationQr } from '../../../../../utils';
 
 const BrightConnectionModal = () => {
-  const userProfile = useContext(UserProfileContext);
+  const { userProfile, refreshUserProfile, loading } = useContext(UserProfileContext);
   const verificationUrl = useMemo(() => userProfile?.verificationUrl || '', [userProfile]);
   const verificationQr = userProfile ? getVerificationQr(userProfile) : '';
   const copyVerificationUrl = () => {
     navigator.clipboard.writeText(verificationUrl);
     alert('Copied');
   };
+
+  const refreshConnectionButtonAction = useCallback(async () => {
+    try {
+      if (refreshUserProfile) {
+        const refreshedUserProfile = await refreshUserProfile();
+        if (refreshedUserProfile) {
+          (refreshedUserProfile.verificationStatus === BrightIdVerificationStatus.VERIFIED) ?
```

Comment:
Could be refreshedUserProfile?.verificationStatus instead of the if the statement above it.

Reply:
I think it's better if we always return the UserProfile object from the callback in the context.